### ISSUE TITLE
Please document / implement securing routes by JWT scope

### ISSUE THREAD

**Issue by jcassee** [micronaut-projects/micronaut-security#254]
Micronaut supports JSON Web Tokens as an authentication mechanism. Often, a JWT was obtained from OAuth 2 and represents a client performing a request on behalf of a user. The ["scope" claim](https://tools.ietf.org/id/draft-bertocci-oauth-access-token-jwt-00.html#AuthorizationClaims) is used to tell the server what authorizations the user has entrusted the client with.

Take for example, an API representing a book store to which I have an account. Although I myself might have the right to buy books through the API, I might give my mobile app only the authorization to browse books. Think also how GitHub third-party apps request certain permissions.

Note also that scopes are different from roles. Although I have the role of customer at the book store, the mobile app is still restricted to only browse books for me. Issue #13 seemed to move in a direction where the two are conflated.

I'm sure there is a way of creating an annotation processor for securing routes by scope, based on `SecuredAnnotationRule` as an example. Because protecting by JWT is (in my experience) a common reason for using JWTs in the first place, it would be useful if it were supported directly (or, if there is already a way, documented).

**Comment by jcassee** [comment]
This is a skeleton implementation of using JWT scope for securing routes:

https://gist.github.com/jcassee/8ce917387263d40b9908506af5b5c578

**Comment by sdelamo** [comment]
Yes, currently we only support `roles` but this is something we could do in the future.

**Comment by traycho** [comment]
I have something similar in this context.

https://github.com/micronaut-projects/micronaut-security/issues/256#issuecomment-644682219

**Comment by dstepanov** [comment]
I think this one can be also solved by custom `RolesFinder`.

**Comment by jcassee** [comment]
@dstepanov I would not mix roles and scopes, I think they really are different things.

**Comment by frehov** [comment]
sorry to necro this issue.

but it's possible to remap the `RolesFinder` to fetch the "roles" from the "scope" claim/attribute by using `micronaut.security.token.roles-name=scope` and then you can use `@Rolesallowed` annotation with the entries from the scope attribute instead. 
This only works if you're not including any roles in the claims as it directly changes what attribute is used to map roles from.
This also seems to have been available in micronaut since version 1.0.0.
